app/src/main/java/com/mk/tasky/core/presentation/TaskyTextField.kt

```diff
@@ -0,0 +1,157 @@
+    trailingIconColor: Color? = null,
+    onIconClick: () -> Unit = {},
+    isPassword: Boolean = false,
+    isValid: Boolean = false,
+    modifier: Modifier = Modifier
+) {
+    var isPasswordHidden by remember {
+        mutableStateOf(true)
+    }
+    OutlinedTextField(
+        modifier = modifier,
+        value = value,
+        onValueChange = onValueChange,
+        keyboardOptions = KeyboardOptions(
+            keyboardType = keyboardType,
+            imeAction = imeAction
+        ),
+        colors = TextFieldDefaults.textFieldColors(
+            errorCursorColor = Red,
+            errorIndicatorColor = Red,
+            errorLabelColor = Red,
+            textColor = DarkGray,
+            placeholderColor = Gray,
+            leadingIconColor = Gray,
+            backgroundColor = Light2,
+            focusedIndicatorColor = LightBlue,
+            unfocusedIndicatorColor = Color.Transparent,
+            trailingIconColor = if (isValid) Green else trailingIconColor ?: Gray,
+            cursorColor = DarkGray
+        ),
+        placeholder = {
+            Text(text = placeholder)
+        },
+        isError = showError,
+        maxLines = maxLines,
+        singleLine = maxLines == 1,
+        trailingIcon = {
+            TrailingIcon(
+                isPassword = isPassword,
+                trailingIcon = trailingIcon,
+                onIconClick = onIconClick,
+                isValid = isValid,
+                onPasswordClick = {
+                    isPasswordHidden = !isPasswordHidden
+                },
+                isPasswordHidden = isPasswordHidden
+            )
+        },
+        shape = RoundedCornerShape(10.dp),
+        visualTransformation = if (isPassword && isPasswordHidden) PasswordVisualTransformation() else VisualTransformation.None
+    )
+}
+
+@Composable
+fun TaskyEmailTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String = "",
+    showError: Boolean = false,
+    isValid: Boolean = false,
+    modifier: Modifier = Modifier
+) {
+    TaskyTextField(
+        value = value,
+        onValueChange = onValueChange,
+        placeholder = placeholder,
+        showError = showError,
+        modifier = modifier,
+        keyboardType = KeyboardType.Email,
+        isValid = isValid
+    )
+}
+
+@Composable
+fun TaskyPasswordTextField(
+    value: String,
+    onValueChange: (String) -> Unit,
+    placeholder: String = "",
+    showError: Boolean = false,
+    isValid: Boolean = false,
+    modifier: Modifier = Modifier
+) {
+    TaskyTextField(
+        value = value,
+        onValueChange = onValueChange,
+        placeholder = placeholder,
+        showError = showError,
+        modifier = modifier,
+        keyboardType = KeyboardType.Password,
+        isPassword = true,
+        isValid = isValid
+    )
+}
+
+@Composable
+private fun TrailingIcon(
+    isPassword: Boolean = false,
+    isPasswordHidden: Boolean,
```

Comment:
If `isPassword` has a default value, this should have too, right? 😄

Reply:
You are right